### PR TITLE
Add loading animation for preview search results.

### PR DESCRIPTION
**Pull request by cdragos**

### CODE REVIEW

File: src/results.js [hunk 1]
```diff
@@ -171,8 +194,8 @@ class Results extends React.Component {
     } else {
       results = <p>-- no results --=</p>
     }
-
     let preview = (this.state || {}).preview
+    let preview_url = preview && url.resolve(window.location.href, preview);
```
Comment (gabriel-v) [inline src/results.js]: The trailing `;` disturbs the dragon.

File: src/results.js [hunk 2]
```diff
@@ -6,11 +6,34 @@ function timeMs() {
...
+    this.loadHandler = this.loadHandler.bind(this)
+  }
+
+  loadHandler() {
+    this.setState({loaded: true})
+  }
+
+  render() {
+    let embedUrl = `${this.props.url}?embed=on`
+    let style = (!this.state.loaded && {display: 'none'}) || {}
```
Comment (mgax) [inline src/results.js]: I think you can write this as `let style = {dislpay: this.state.loaded}` and react will compile it to the correct inline css.

File: src/results.js [hunk 3]
```diff
@@ -6,11 +6,34 @@ function timeMs() {
...
+
+    return (
+      <div key={this.props.url}>
+        <iframe className='results-item-preview'
+                onLoad={this.loadHandler} src={embedUrl}
```
Comment (mgax) [inline src/results.js]: Maybe it's easier to just inline the callbcak, so you don't have to bind it:
```javascript
<iframe onLoad={() => { this.setState({loaded: true}) }} />
```

File: src/results.js [hunk 4]
```diff
@@ -6,11 +6,34 @@ function timeMs() {
...
-function Preview({url}) {
-  let embedUrl = `${url}?embed=on`
-  return (
-    <iframe className='results-item-preview' src={embedUrl} />
-  )
+class Preview extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {loaded: false}
```
Comment (mgax) [inline src/results.js]: You can do it simpler: don't set `this.state` in constructor, and below, accept the fact that `this.state` might be undefined:
```javascript
(this.state || {}).loaded // at first this evaluates to undefined, then to true
```

File: src/results.js (hunk 1 above)
Comment (mgax) [inline src/results.js]: Let's use JavaScript naming conventions: `previewUrl`.

**Comment by mgax** [comment]
I think our CSS is getting complicated. Maybe it's time to use a preprocessor (e.g. scss) and split it into multiple files.

File: src/results.js (hunk 1 above)
Comment (cdragos) [inline src/results.js]: ES5, ES2015, ES6+. Too much noise...

File: src/results.js (hunk 2 above)
Comment (mgax) [inline src/results.js]: You're right, I was confusing this with the api from classnames, which does something else anyway.

**Pull request merged**